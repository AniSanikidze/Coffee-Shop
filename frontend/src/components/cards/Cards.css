.products-container{
  display: flex;
  margin: 2vmax auto 0;
  width: 80vw;
  flex-wrap: wrap;
  justify-content: center;
  max-width: 100%;
}


.productCard {
  width: 14vmax;
  display: flex;
  flex-direction: column;
  text-decoration: none;
  color: rgb(48,48,48);
  margin:2vmax;
  transition: all 3s;
  padding-bottom: 0.5vmax;
  align-items: center;
  font-family: 'firago';
}

.productCard > img {
  width: 14vmax;
  /* height: 21vmax; */
}

.productCard > div {
  margin: 0.5vmax;
  display: flex;
  justify-content: flex-start;
  align-items: center;
}

.productCard > p {
  /* font-family: "Roboto"; */
  font-size: 1.4vmax;
  margin: 1vmax 0.5vmax;
  margin-bottom: 0;
  text-transform: uppercase;
  color:#555555 !important;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  /* font-weight: bold; */
}

.productCart > img{
  width: 14vmax !important;
}

.add-to-cart {
  display: none !important;
  margin-top: 0;
    opacity: 1;
    -ms-transform: translateY(0);
    -webkit-transform: translateY(0);
    transform: translateY(0);
    width: 100%;
    left: 0;
    vertical-align: middle;
    font: inherit;
    text-align: center;
    margin: 0;
    cursor: pointer;
    font-family: Montserrat,sans-serif;
    letter-spacing: 1px;
    text-transform: uppercase;
    transition: all .3s ease-in-out;
    font-size: 13px;
    font-weight: 600;
    font-style: normal;
    height: auto;
    line-height: 30px;
    padding: 9px 30px;
    border-radius: 0;
    background-color: #afa483;
}

.productCard > span {
  margin: 0.5vmax;
  color: #555555;
  font-family: "Franklin Gothic Medium", "Arial Narrow", Arial, sans-serif;
  font-size: 1.1vmax;
}

.productCard .btn {
  border: 2px solid #555555;
}

.productCard .btn:hover {
  background-color: #f0f0f0;
}

.productCard:hover {
  box-shadow: 0 0 5px rgba(15, 15, 15, 0.26);
  transform: translateY(-1vmax);
}

.productCard:hover .add-to-cart {
  position: relative;
  display: block !important;
  margin-top: 0;
    opacity: 1;
    visibility: visible;
    -ms-transform: translateY(0);
    -webkit-transform: translateY(0);
    transform: translateY(2);
    width: 100%;
    left: 0;
}



@media screen and (max-width: 1300px) {
.productCard > div{
  display: flex;
  flex-direction: column;
}

}

@media screen and (max-width: 600px) {
  .productCard > p {
    font-size: 1.7vmax;
  }

  .productCard > div {
    margin: 0vmax;
    display: block;
  }

  .productCard > span {
    font-size: 1.5vmax;
  }

  .productCard > div > span {
    margin: 0 0.5vmax;
    font: 300 1vmax "Roboto";
  }
}


.cards {
  padding: 2rem 4rem 0 4rem;
  background: #fff;
}

.cards h1 {
  text-align: center !important;
  border-bottom: 1px solid rgba(21, 21, 21, 0.5);
  width: 20vmax;
  padding: 1vmax;
  margin: auto;
  color:#555555;
}

.cards__container {
  display: flex;
  flex-flow: column;
  align-items: center;
  max-width: 1120px;
  width: 90%;
  margin: 0 auto;
}

.cards__wrapper {
  position: relative;
  margin: 50px 0 20px;
}

.cards__items {
  margin-bottom: 24px;
}

.cards__item {
  display: flex;
  flex: 1;
  margin: 0 1rem;
  border-radius: 10px;
}

.cards__item__link {
  display: flex;
  flex-flow: column;
  width: 100%;
  box-shadow: 0 6px 20px rgba(56, 125, 255, 0.17);
  -webkit-filter: drop-shadow(0 6px 20px rgba(56, 125, 255, 0.017));
  filter: drop-shadow(0 6px 20px rgba(56, 125, 255, 0.017));
  border-radius: 10px;
  overflow: hidden;
  text-decoration: none;
}

.cards__item__pic-wrap {
  position: relative;
  width: 100%;
  padding-top: 67%;
  overflow: hidden;
  z-index: 1;
}

.cards__item__pic-wrap::after {
  content: attr(data-category);
  position: absolute;
  bottom: 0;
  margin-left: 10px;
  padding: 6px 8px;
  max-width: calc((100%) - 60px);
  font-size: 12px;
  font-weight: 700;
  color: #fff;
  background-color: #1f98f4;
  box-sizing: border-box;
}

.cards__item__img {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: block;
  width: 100%;
  max-width: 100%;
  height: 100%;
  max-height: 100%;
  object-fit: cover;
  transition: all 0.2s linear;
  z-index: -1;
  overflow: hidden;
}

.cards__item__img:hover {
  transform: scale(1.1);
}

.cards__item__info {
  padding: 20px 30px 30px;
}

.cards__item__text {
  color: #252e48;
  font-size: 18px;
  line-height: 24px;
}

@media only screen and (min-width: 1200px) {
  .content__blog__container {
    width: 84%;
  }
}

@media only screen and (min-width: 1024px) {
  .cards__items {
    display: flex;
  }
}

@media only screen and (max-width: 1024px) {
  .cards__item {
    margin-bottom: 2rem;
  }
}