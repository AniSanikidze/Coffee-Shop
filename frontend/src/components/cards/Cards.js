import React,{useEffect,useContext} from 'react';
import Aos from 'aos'
import 'aos/dist/aos.css';
import CardItem from './CardItem';
// import { TopSuggestionsData } from './TopSuggestionsData'
import './Cards.css';
// import Product from '../products/Product';
import {clearErrors, getProduct} from '../../actions/productAction'
import {useSelector, useDispatch} from 'react-redux'
import Loader from '../loading/Loader';
import { useAlert } from 'react-alert';
import { UserContext } from '../../UserContext';
import Searchbox, {input} from '../search/Searchbox'

function Cards({match}) {
  // const { searchInput } = useContext(UserContext);
  const alert = useAlert()
  const dispatch = useDispatch()
  let {loading,error,products} = useSelector(state=>state.products)
  const { input, setInput } = Searchbox();

  useEffect(() => {
    if(error){
      alert.error(error)
      dispatch(clearErrors())
    }
    dispatch(getProduct())
    setFeaturedCoffee()
  },[dispatch,error,input])

  function setFeaturedCoffee () {
    switch(window.location.pathname){
      case '/':
        products = products.slice(0, 4)
         return products
      default:
        return products;
    }
  }

  // useEffect(() => {
  //     setFeaturedCoffee();
  // },[])

  setFeaturedCoffee();
  
  // console.log(match)
  

  // const a = useEffect(() => {
  //   Aos.init({ duration: 1000 })
  // }, [])
  
  // window.addEventListener('a', a);
  // console.log(useSelector(state => state.products))

  return (
    <div className='cards'>
      <h1>Featured Products </h1>
      <div className='products-container'>
      {loading ? <Loader/> : 
      products && products.map(product => (
        // console.log(product)
      <CardItem productName={product.productName}
      id={product._id}
      price={product.price}
      numOfReviews={product.numOfReviews}
      stock={product.stock} 
      rating={product.rating}
      // category={product.category}
      />
      ))}
      </div><div className='cards__container'>
        {/* <div className='cards__wrapper'>
          <ul className='cards__items'>
            {/* {TopSuggestionsData.map(suggestion => {
              return TopSuggestionsData.indexOf(suggestion) < 2 &&
                      <CardItem
                        src={suggestion.src}
                        text={suggestion.text}
                        label={suggestion.label}
                        filterValue={suggestion.filterValue}
                      />
            })} */}
          {/* </ul> */}
          {/* <ul className='cards__items'>
            {TopSuggestionsData.map(suggestion => {
              return TopSuggestionsData.indexOf(suggestion) >= 2 &&
                      <CardItem
                        src={suggestion.src}
                        text={suggestion.text}
                        label={suggestion.label}
                        filterValue={suggestion.filterValue}
                      />
            })}
          </ul>
        </div> */}
      </div>
    </div>
  );
}

export default Cards;