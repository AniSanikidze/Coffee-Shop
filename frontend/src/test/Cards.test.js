import * as React from "react";
import * as ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom'
import Cards from "../components/cards/Cards";
import { UserContext } from "../UserContext";

// test('renders h1', () => {
//     const root = document.createElement("div");
//     ReactDOM.render(<Cards />, root);
//     expect(root.querySelector("h1").textContent).toBe("Check Out the Top Suggestions! ");
// })

describe('testing Cards', () => {

    it('should render', () => {
        const CardItemComponent = renderer.create(<UserContext.Provider value={false}><Router><Cards /></Router></UserContext.Provider>).toJSON();
        expect(CardItemComponent).toMatchSnapshot();
    })

})