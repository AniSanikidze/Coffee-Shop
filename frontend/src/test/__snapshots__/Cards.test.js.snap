// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Cards should render 1`] = `
<div
  className="cards"
>
  <h1>
    Check Out the Top Suggestions! 
  </h1>
  <div
    className="cards__container"
  >
    <div
      className="cards__wrapper"
    >
      <ul
        className="cards__items"
      >
        <li
          className="cards__item"
          onClick={[Function]}
        >
          <a
            className="cards__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="International"
            >
              <img
                alt="Food Category"
                className="cards__item__img"
                src="images/Home/international.webp"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                View the city's more than 1000 International restaurants
              </h5>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
          onClick={[Function]}
        >
          <a
            className="cards__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Italian"
            >
              <img
                alt="Food Category"
                className="cards__item__img"
                src="images/Home/italian.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Explore all the Italian restaurants of Prague
              </h5>
            </div>
          </a>
        </li>
      </ul>
      <ul
        className="cards__items"
      >
        <li
          className="cards__item"
          onClick={[Function]}
        >
          <a
            className="cards__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Czech"
            >
              <img
                alt="Food Category"
                className="cards__item__img"
                src="images/Home/czech.jpg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Find your most suitable Czech restaurant
              </h5>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
          onClick={[Function]}
        >
          <a
            className="cards__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Gluten Free"
            >
              <img
                alt="Food Category"
                className="cards__item__img"
                src="images/Home/glutenfree.webp"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Check out  the restaurants with gluten free options
              </h5>
            </div>
          </a>
        </li>
        <li
          className="cards__item"
          onClick={[Function]}
        >
          <a
            className="cards__item__link"
            href="/restaurants"
            onClick={[Function]}
          >
            <figure
              className="cards__item__pic-wrap"
              data-category="Vegetarian"
            >
              <img
                alt="Food Category"
                className="cards__item__img"
                src="images/Home/vegan.jpeg"
              />
            </figure>
            <div
              className="cards__item__info"
            >
              <h5
                className="cards__item__text"
              >
                Check out the restaurants with vegetarian meals
              </h5>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
